web/static/js/components/folders/FolderShow.jsx

```diff
@@ -104,25 +121,50 @@ class FolderShow extends Component<any, any> {
 
     const readOnly = !project || project.readOnly
 
-    let addButton = null
+    let primaryButton = null
     if (!readOnly) {
-      addButton = (
-        <AddButton text={t('Add survey')} onClick={() => this.newSurvey()} />
-      )
+      if (panelSurvey) {
+        primaryButton = (
+          <RepeatButton text={t('Repeat survey')} disabled={panelSurvey.isRepeatable} onClick={() => this.repeatSurvey()} />
+        )
+      } else {
+        primaryButton = (
+          <AddButton text={t('Add survey')} onClick={() => this.newSurvey()} />
+        )
+      }
+    }
+
+    const emptyFolder = surveys && surveys.length == 0
+    if (panelSurvey && emptyFolder) {
+      throw new Error(t('Empty panel survey'))
+    }
+
+    let hint = null
+    if (panelSurvey) {
+      hint = <div className='repeat-survey row'>
+        <div className='col s12 hint'>
+          {
+            panelSurvey.isRepeatable
+            ? t('The last occurence of the panel survey is complete, you may follow up with a new survey sent to a subset of the respondents of this panel survey')
+            : t("The last occurence of the panel survey isn't complete yet. After that, you may follow up with a new survey sent to a subset of the respondents of this panel survey")
+          }
+        </div>
+      </div>
```

Comment:
same as `primaryButton`. I would extract all this logic to a simple and dedicated function